core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/plangen/LogicalPlan.java

```diff
@@ -514,25 +516,31 @@ private void handleSetInputOperator(IOperator src, IOperator dest) throws PlanGe
             }  
         }
     }
-     
+
     /*
-     * Finds the sink operator in the operator graph.
-     * 
-     * This function assumes that the graph is valid and there is only one sink in the graph.
+     * Finds all sink operators in the operator graph.
+     * his function assumes that the graph is valid and there is at least one sink in the graph.
      */
-    private ISink findSinkOperator(HashMap<String, IOperator> operatorObjectMap) throws PlanGenException {
-        IOperator sinkOperator = adjacencyList.keySet().stream()
-                .filter(operator -> operatorPredicateMap.get(operator)
-                        .getClass().toString().toLowerCase().contains("sink"))
-                .map(operator -> operatorObjectMap.get(operator))
-                .findFirst().orElse(null);
-        
-        PlanGenUtils.planGenAssert(sinkOperator != null, "Error: sink operator doesn't exist.");
-        PlanGenUtils.planGenAssert(sinkOperator instanceof ISink, "Error: sink operator's type doesn't match.");
-        
-        return (ISink) sinkOperator;
+    private HashMap<String, ISink> findSinkOperators(HashMap<String, IOperator> operatorObjectMap) throws PlanGenException {
+
+        HashMap<String, IOperator> operatorMap = new HashMap<>();
+        for (HashMap.Entry<String, IOperator> entry: operatorObjectMap.entrySet()) {
+            if (entry.getValue().getClass().toString().toLowerCase().contains("sink")) {
```

Comment:
let's replace checking class name to use `ISink` class to check: `operator instanceof ISink`. Also make sure all of our sink operators are implementing the `ISink` interface.

Reply:
I will fix it.